### ISSUE TITLE
Electrostatics key mismatch after combination

### ISSUE THREAD

**Issue by mattwthompson** [openforcefield/openff-interchange#935]
@j-wags and I have a use case that's failing with a cryptic `KeyError` when exporting to OpenMM; last week I thought I isolated it as a type mismatch in `ElectrostaticsHandler.key_map` (https://github.com/openforcefield/openff-interchange/pull/932) but I can't reproduce it. I expected something like this to trigger it:

```python
import openmm.app
import openmm.unit
from openff.toolkit import ForceField, Quantity, Topology
from openmmforcefields.generators import SMIRNOFFTemplateGenerator

from openff.interchange import Interchange
from openff.interchange._tests import MoleculeWithConformer
from openff.interchange.drivers import get_openmm_energies

benzene = MoleculeWithConformer.from_smiles("c1ccccc1")
hexanol = MoleculeWithConformer.from_smiles("CCCCCCO")
hexanol._conformers[0] += Quantity([5, 5, 5], "angstrom")

topology = Topology.from_molecules([benzene, hexanol])
topology.box_vectors = Quantity([4, 4, 4], "nanometer")

for molecule in topology.molecules:
    molecule.assign_partial_charges(partial_charge_method="am1bcc")

forcefield = openmm.app.ForceField()
forcefield.registerTemplateGenerator(
    SMIRNOFFTemplateGenerator(molecules=list(topology.molecules)).generator,
)

system = forcefield.createSystem(
    topology=topology.to_openmm(),
    nonbondedMethod=openmm.app.PME,
    nonbondedCutoff=0.9 * openmm.unit.nanometer,
    constraints=openmm.app.HBonds,
    switchDistance=0.8 * openmm.unit.nanometer,
)

imported = Interchange.from_openmm(
    system=system,
    topology=topology.to_openmm(),
    positions=topology.get_positions().to_openmm(),
)

print(
    get_openmm_energies(
        ForceField("openff_unconstrained-2.0.0.offxml").create_interchange(topology),
    ),
)

# this should fail with a KeyError
print(get_openmm_energies(imported))

```
but it's just not

**Comment by mattwthompson** [comment]
I can reproduce it with this:

```python
import openmm.app
import openmm.unit
from openff.toolkit import ForceField, Quantity, Topology

from openff.interchange import Interchange
from openff.interchange._tests import MoleculeWithConformer

benzene = MoleculeWithConformer.from_smiles("c1ccccc1")
hexanol = MoleculeWithConformer.from_smiles("CCCCCCO")
hexanol._conformers[0] += Quantity([5, 5, 5], "angstrom")

topology = Topology.from_molecules([benzene, hexanol])
topology.box_vectors = Quantity([6, 6, 6], "nanometer")

popc = MoleculeWithConformer.from_smiles(
    "CCCCCCCCCCCCCCCC(=O)OCC(COP(=O)([O-])OCC[N+](C)(C)C)OC(=O)CCCCCCCC=CCCCCCCCC",
    allow_undefined_stereo=True,
)

popc_topology = popc.to_topology()
popc_topology.box_vectors = Quantity([6, 6, 6], "nanometer")

forcefield = openmm.app.ForceField("amber/lipid17.xml")

popc_system = forcefield.createSystem(
    topology=popc_topology.to_openmm(),
    nonbondedMethod=openmm.app.PME,
    nonbondedCutoff=0.9 * openmm.unit.nanometer,
    constraints=openmm.app.HBonds,
    switchDistance=0.8 * openmm.unit.nanometer,
)

imported = Interchange.from_openmm(
    system=popc_system,
    topology=popc_topology,
    positions=popc_topology.get_positions(),
)

combined = imported.combine(
    ForceField("openff_unconstrained-2.0.0.offxml").create_interchange(topology),
)

# Runs fine
imported.to_openmm()

# Fails!
combined.to_openmm()
"""
Traceback (most recent call last):
  File "/Users/mattthompson/software/openff-interchange/run.py", line 47, in <module>
    combined.to_openmm()
  File "/Users/mattthompson/software/openff-interchange/openff/interchange/components/interchange.py", line 545, in to_openmm_system
    return _to_openmm_system(
  File "/Users/mattthompson/micromamba/envs/openff-interchange-dev/lib/python3.10/site-packages/openff/utilities/utilities.py", line 80, in wrapper
    return function(*args, **kwargs)
  File "/Users/mattthompson/software/openff-interchange/openff/interchange/interop/openmm/__init__.py", line 92, in to_openmm_system
    particle_map = _process_nonbonded_forces(
  File "/Users/mattthompson/software/openff-interchange/openff/interchange/interop/openmm/_nonbonded.py", line 147, in _process_nonbonded_forces
    _func(
  File "/Users/mattthompson/software/openff-interchange/openff/interchange/interop/openmm/_nonbonded.py", line 379, in _create_single_nonbonded_force
    partial_charge = partial_charges[top_key].m_as(unit.e)
KeyError: TopologyKey with atom indices (134,)
"""
```

It requires combination, I haven't yet isolated if it requires `from_openmm`

**Comment by mattwthompson** [comment]
Probably fixed by #932

**Comment by mattwthompson** [comment]
Haven't seen reports otherwise
